src/tools/pub/automated-publishing.md

```diff
@@ -0,0 +1,719 @@
+[build configuration file schema][15].
+
+A Cloud Build configuration file that works with this example should
+resemble the following YAML markup:
+
+```yaml
+# cloudbuild.yaml
+steps:
+- id: Create temporary token
+  name: gcr.io/cloud-builders/gcloud
+  volumes:
+  - name: temporary-secrets
+    path: /secrets
+  script: |
+    gcloud auth print-identity-token \
+      --impersonate-service-account=pub-dev@$PROJECT_ID.iam.gserviceaccount.com \
+      --audiences=https://pub.dev \
+      --include-email > /secrets/temporary-pub-token.txt
+  env:
+  - PROJECT_ID=$PROJECT_ID
+- id: Publish to pub.dev
+  name: dart
+  volumes:
+  - name: temporary-secrets
+    path: /secrets
+  script: | 
+    cat /secrets/temporary-pub-token.txt | dart pub token add https://pub.dev
+    dart pub publish --force
+```
+
+The `Create temporary token` step features the
+`gcloud auth print-identity-token` script.
+This script performs the following actions:
+
+* It creates an OpenID Connect `id_token`.
+  Google signs this `id_token`. The signature that expires within one
+  hour.
+* The `--impersonate-service-account` option identifies the service
+  account to impersonate. This example uses the service account created
+  in a previous activity in this guide.
+* The `--audiences` option identifies `pub.dev` as the intended
+  recipient of the token. 
+* The `--include-email` option requests that `pub.dev` recognize the
+  service account.
+
+Once you create the `id_token`, Google Cloud writes it to a file that
+resides in a _volume_. Google Cloud uses this mechanism to
+[pass data between steps][16].
+Don't store the token in `/workspace`.
+Google Cloud checks out your repository into `/workspace`.
+If you store the token in `/workspace`, your published package includes
+the `id_token`.
+
+
+### Creating a Cloud Build trigger
+
+With a service account configured and a `cloudbuild.yaml` file in
+the repository, use [console.cloud.google.com](https://console.cloud.google.com/cloud-build/triggers)
+dashboard to create a _Cloud Build Trigger_.
+To create a build trigger, you need to connect to a _source repository_
+and specify which events should trigger a build. You can use [GitHub][18],
+[Cloud Source Repository][19], or one of the [other options][20].
+To learn how to configure a _Cloud Build Trigger_, check out
+[creating and managing build triggers][21].
+
+To publish packages using the Cloud Build Configuration file you created:
+
+1. Open the **Triggers** page in the Google Cloud console.
+1. Select your project from the project selector drop-down menu at the
+   top of the page.
+1. Click **Open**.
+1. If you have an existing trigger, open it. 
+   If you need a new trigger, [create it][28].
+1. Scroll to the **Configuration** section.
+1. Click **Cloud Build Configuration**.
+
+    ![Configuration for trigger](gcb-trigger-configuration.png)
+
+1. Do **not** specify a **Service account** to trigger the build.
+   Let Cloud use its default service account.
```

Comment:
I don't think guide is a good idea.

We're not going to provide explicit steps for this.
There are many ways you can connect to a source repository.

Many events you can configure, etc.

We're not going to debate these in detail. We could pick one and give an example, but then we'd be going into details about how to use Cloud Build with a specific kind of source repository. I think this is unnecessarily detailed.

Let's instead say that you must configure "Cloud Build Configuration" and not specify a service account.

The cloud build configuration is really obvious, specifying a service account is the only tempting mistake you can make because we've previously created a service account.
But it was created to be impersonated by the default Cloud Build service account, not to be used as service account for cloud build runs, this is a distinction that we should explain. Because it's an obvious mistake most people include your truly would make :D

Reply:
IMO, I think it's best to leave it up to `[creating and managing build triggers][21]` to explain how to create a trigger